device/pcie/pci_device.cpp

```diff
@@ -687,6 +687,14 @@ tt::umd::architecture_implementation *PCIDevice::get_architecture_implementation
 bool PCIDevice::init_hugepage(uint32_t num_host_mem_channels) {
     const size_t hugepage_size = HUGEPAGE_REGION_SIZE;
 
+    if (numa_node > numa_max_node()) {
+        log_warning(LogSiliconDriver, "numa_node: {} is greater than numa_max_node: {}.", numa_node, numa_max_node());
+    }
+
+    // Prefer allocations on the NUMA node associated with the device.
+    numa_set_preferred(numa_node);
```

Comment:
Can you please create a tt_metal branch which consumes this UMD commit, and then run tt_metal CI? We don't have machines on our CI that are affected by the change in this PR.

You could run these:
    "all-post-commit-workflows.yaml"
    "t3000-unit-tests.yaml"
    "tg-unit-tests.yaml"
    "tgg-unit-tests.yaml"

Reply:
This exposed deficiencies in Metal CI, which is running tests that request more hugepages than are actually provisioned.